### PR TITLE
Introducing Modals to Crayons 🤩

### PR DESCRIPTION
**Pull request by asifahmedfw**
You can now use modals in your apps via Crayons. Modals are higly customizable overlays to help you
show data to your users in a focused way.

re #113

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes. (WIP)
- [x] All new and existing tests passed.
- [x] My commits have standard messages as mentioned in [Contributing Guidelines](./../blob/next/CONTRIBUTING.md)
 
## How Has This Been Tested?
This has been tested locally with Storybook

### CODE REVIEW

**Comment by shravan-balasubramanian** [comment]
Hey @asifahmedfw. Will this make success and failure buttons optional now. Just to display plain modals?

**Comment by shravan-balasubramanian** [comment]
Hey @asifahmedfw. Will this make success and failure buttons optional now. Just to display plain modals?

File: src/components.d.ts [hunk 1]
```diff
@@ -19,6 +19,10 @@ export namespace Components {
           * Sets the button to a full-width block. If the attribute’s value is undefined, the value is set to false.
          */
         "expand": boolean;
+        /**
+          * Accepts the id of the fw-modal component to open it on click
+         */
+        "modalTriggerId": string;
```
Comment (shravan-balasubramanian) [inline src/components.d.ts:25]: Can we also expose `openModal` method for the cases where a user wants to open up a modal without a button in the picture

File: src/components/modal/modal.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    return <fw-icon class="icon" name={this.icon} size={16}></fw-icon>;
+  }
+
+  renderFooter() {
+    return this.customFooter
+      ? <slot name="footer"/>
+      : (
+      <span>
+        <fw-button color="secondary" onClick={() => this.closeModal()}>{this.cancelText}</fw-button>
+        <fw-button color="primary">{this.successText}</fw-button>
```
Comment (shravan-balasubramanian) [inline src/components/modal/modal.tsx:63]: Can we emit events for both Ok and Cancel button clicks

Comment (asifahmedfw) [reply src/components/modal/modal.tsx:63]: Will do so. 😄

File: src/components.d.ts (hunk 1 above)
Comment (asifahmedfw) [reply src/components.d.ts:25]: It can be done through the visible property in the modal. Setting visible to true will show it.

**Comment by asifahmedfw** [comment]
> Hey @asifahmedfw. Will this make success and failure buttons optional now. Just to display plain modals?

Will add a `no-footer` attribute.

File: src/components/modal/modal.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+describe('fw-modal', () => {
+  it('renders', async () => {
+    const page = await newE2EPage();
+
+    await page.setContent('<fw-modal></fw-modal>');
+    const element = await page.find('fw-modal');
+    expect(element).toHaveClass('hydrated');
+  });
+});
```
Comment (shravan-balasubramanian) [inline src/components/modal/modal.e2e.ts:11]: Can we capture tests for events and other edge cases props

Comment (asifahmedfw) [reply src/components/modal/modal.e2e.ts:11]: WIP as mentioned above. Will update it very soon 😉

**Review by shravan-balasubramanian — APPROVED** [review]
Thanks!
